amazon/ion/ioncmodule.c

```diff
@@ -372,6 +372,34 @@ static iERR ionc_write_sequence(hWRITER writer, PyObject* sequence, PyObject* tu
     Py_DECREF(sequence);
     cRETURN;
 }
+/*
+ * Process and write the key-value pair.
+ *  *  Args:
+ *      writer:  An ion writer
+ *      key: The key of IonStruct item
+ *      val: The value of IonStruct item
+ *      tuple_as_sexp: Decides if a tuple is treated as sexp
+ */
+
+static iERR write_key_value(hWRITER writer, PyObject* key, PyObject* val, PyObject* tuple_as_sexp) {
+    iERR err;
+    if (PyUnicode_Check(key)) {
+        ION_STRING field_name;
+        ion_string_from_py(key, &field_name);
+        IONCHECK(ion_writer_write_field_name(writer, &field_name));
+    } else if (key == Py_None) {
+        IONCHECK(_ion_writer_write_field_sid_helper(writer, 0));
```

Comment:
Is there not a public API for this? I know we are the developers of ion-c but I don't think we want to be calling `_` prefixed methods from another repo?

Reply:
This concern deserves more investigation. Unfortunately I couldn't find the public API for this functionality in `ion-c`. There are several spots using the `_` prefixed methods in [ionc_write_symboltoken](https://github.com/amazon-ion/ion-python/blob/d425587d94f8e9fa22368ed1b82a1b65fcb81820/amazon/ion/ioncmodule.c#L278C18-L278C18). I'm considering whether we should propose a new strategy for these implementations that avoids using the `_` prefixed methods, and then update them all at once.

Reply:
Well ok, but that's not great. I also see this pre-existed...

But really this is papering over a potential data correctness issue when roundtripping Ion with unresolved imports. Should probably just throw here. I know that's technically a "breaking change" but we should've just thrown here in the first place (aside from implementing the round-tripping correctly).

Reply:
... on further consideration:
There is a correctness issue with roundtripping, but that needs to be handled on the read side.

The way to do this without calling protected/private methods is to construct an `ION_SYMBOL` that has null text and SId of 0 and no import location. Of course this is pre-existing so you're not obligated to.

Reply:
Thanks for the explanation.